src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -528,6 +564,10 @@ private class MacroResolver private constructor(
             return true
         }
 
+        (startElement.contextOrSelf<RsElement>()?.crateRoot as? RsFile)?.let { crateRoot ->
```

Comment:
Let's extract `startElement.contextOrSelf<RsElement>()?.crateRoot as? RsFile` into local variable and check for null with simple `if`
I think it will be a little bit readable (at least for me)

Reply:
This code is related to `org.rust.lang.core.resolve.RsMacroResolveTest#test macro_export macro is visible in the same crate without macro_use` test, isn't it? If yes, maybe it's worth to add special testmark for this case?